### ISSUE TITLE
Improve squashed commits

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#1575]
In this repo, we adhere to the `conventional commits` convention to write nice, machine-readable, commits that can be reused to create change-logs, bump the semantic version of the package and so on.

Our current workflow is: code locally, open a GitHub PR and merge the PR from the GH user interface.
The problem lies in this last part: we can't force a conventional commit style into the merging commit.

This causes a set of unwanted problems:
- we forgot/misspell the breaking change `BREAKING CHANGE` body
- the description body, used in some cases in the changelog, is the GH default one with the list of all the commits, causing a bad changelog entry
- the PR can be merged with a non `conventional commit` style, causing `semantic-release` to bump the package version without considering that commit.

The possible solutions are the following:
1. create a browser extensions that intercept the title/body of a squashing PR and check/block the PR from being merged if the fields doesn't adhere to the standard.
2. squash the PR from a command line making sure that all the fields are nicely filled.
3. use `pre-receive` hooks in [GH enterprise](https://docs.github.com/en/enterprise-server@3.3/admin/policies/enforcing-policy-with-pre-receive-hooks/about-pre-receive-hooks)

### props/cons:

1. *browser extension*:
    pros: don't change the current PR workflow
    cons: needs to be installed by all devs (at lease within the team), this extension doesn't exist yet, needs to be crafted, but can follow some existing extensions [enhanced-github](https://github.com/refined-github/refined-github) 

2. *CLI tool*
    pros: we already have all the CLI tools to nicely cleanup commits
    cons: we are changing the PR workflow, moving the merge step into a local CLI command

3. *pre-receive-hook*
    pros: don't change the current PR workflow
    cons: we should understand clearly how this hooks works, what kind of message can give back to the user

### Ideas for the solution:

In both solutions we can double-check if the PR is marked with a `breaking change` label and the merge message doesn't contain a `BREAKING CHANGE` section.

#### CLI
We can use a mix of [commitizen](https://github.com/commitizen/cz-cli) + the [github CLI](https://cli.github.com/)
In particular github CLI has now the ability to [submit a squash PR commit title + body](https://github.com/cli/cli/pull/3344) that will play great with the `commitizen` tool.
I'm imaging a command that:
- lists the last x open PRs
- you can select one PR
- enter the PR squash-and-merge mode showing the commitizen form
- when the form ends just send the GH CLI command with the title/body created to merge the PR

#### Browser extension

We can follow the [enhanced-github](https://github.com/refined-github/refined-github) extension. Intercept the text of the PR title/body when clicking on `merge`.
The extension needs to "disable" the merge button if the title/extension doesn't adhere to the standard.

#### Pre-receive hook

We need to check with Infra if that is possible and how this will work with our toolset + GH UI

@nickofthyme @monfera @rshen91 happy to discuss the best way to handle that issue

**Comment by nickofthyme** [comment]
Agreed, I like the idea of a GitHub hook. I asked a question about this in the community forum months ago with no response 😔
